### ISSUE TITLE
Cannot Connect to msn

### ISSUE THREAD

**Issue by NickCis** [emesene/emesene#915]
Sometimes emesene can't connect to the msn network. I leave the terminal output:

newpc@newpc-desktop:~/emesene/emesene$ ./emesene 
/usr/lib/python2.6/dist-packages/xmpp/auth.py:24: DeprecationWarning: the sha module is deprecated; use the hashlib module instead
  import sha,base64,random,dispatcher,re
/usr/lib/python2.6/dist-packages/xmpp/auth.py:26: DeprecationWarning: the md5 module is deprecated; use hashlib instead
  import md5
No handlers could be found for logger "emesene"
[22:57:43 WARNING papyon.client] already logged out
Traceback (most recent call last):
  File "/home/newpc/emesene/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'

**Comment by moneycat** [comment]
Confirm this issue with emesene 2.11.9 recent days. But I think this may an issue of MSN's server if much user encounter.
Here's my log:
$ emesene -v
[09:49:47 INFO papyon.msnp2p.transport] Registering transport <NotificationP2PTransport object at 0x2418780 (papyon+msnp2p+transport+base+BaseP2PTransport at 0x262ce80)>
[09:49:50 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
Traceback (most recent call last):
  File "/usr/share/emesene/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'
...

**Comment by kobergkamp** [comment]
I can connect with ebuddy, but can't with my emesene 1.6.1
I tried to run 2.11.7 on my computer but it just won't work. No response at all after double clicking the button.

**Comment by jrstravino** [comment]
also, this issue with emesene:

[00:29:29 INFO papyon.msnp2p.transport] Registering transport <NotificationP2PTransport object at 0x985a1e4 (papyon+msnp2p+transport+base+BaseP2PTransport at 0x998c170)>
[00:29:35 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
Traceback (most recent call last):
  File "/usr/share/emesene/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'

**Comment by danielgomon** [comment]
I have exactly the same problem, can't connect at all. I have this problem with 2.11.9 and git version.

2.11.9 output:

$ emesene -v
/usr/share/emesene/emesene/e3/jabber/Worker.py:27: DeprecationWarning: the sha module is deprecated; use the hashlib module instead
  import sha
/usr/lib/python2.6/dist-packages/xmpp/auth.py:26: DeprecationWarning: the md5 module is deprecated; use hashlib instead
  import md5
[22:30:55 INFO papyon.msnp2p.transport] Registering transport <NotificationP2PTransport object at 0xa5f61bc (papyon+msnp2p+transport+base+BaseP2PTransport at 0xa721000)>
[22:31:08 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
Traceback (most recent call last):
  File "/usr/share/emesene/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'

git version output:

$ ./emesene/emesene/emesene -v
/usr/lib/python2.6/dist-packages/xmpp/auth.py:24: DeprecationWarning: the sha module is deprecated; use the hashlib module instead
  import sha,base64,random,dispatcher,re
/usr/lib/python2.6/dist-packages/xmpp/auth.py:26: DeprecationWarning: the md5 module is deprecated; use hashlib instead
  import md5
No handlers could be found for logger "emesene"
[22:31:44 INFO extension] extension id /usr/share/emesene/emesene/e3/papylib/Session.py:Session not registered on session
[22:31:44 INFO papyon.msnp2p.transport] Registering transport <NotificationP2PTransport object at 0x94e898c (papyon+msnp2p+transport+base+BaseP2PTransport at 0x95ad9c0)>
[22:31:48 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
[22:31:48 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
[22:31:49 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
[22:31:58 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
[22:31:59 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
[22:32:02 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
[22:32:24 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []

**Comment by Makyui** [comment]
Same with 1.6.3 and 2.11.9 and git in Arch Linux, and Kmess doesn't connect properly either. Sometimes trying to reconnect eventually got a connection but now it doesn't connect at all.

**Comment by gaido** [comment]
Same problem started yesterday, debian amd64, emesene 2.11.9 and older.

$ emesene 
/usr/share/themes/Shiki-Wise/gtk-2.0/gtkrc:126: Murrine configuration option "gradients" is no longer supported and will be ignored.
/usr/share/emesene/emesene/e3/jabber/Worker.py:27: DeprecationWarning: the sha module is deprecated; use the hashlib module instead
  import sha
No handlers could be found for logger "emesene"
/usr/lib/python2.6/dist-packages/Crypto/Util/randpool.py:40: RandomPool_DeprecationWarning: This application uses RandomPool, which is BROKEN in older releases.  See http://www.pycrypto.org/randpool-broken
  RandomPool_DeprecationWarning)
Traceback (most recent call last):
  File "/usr/share/emesene/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'

**Comment by ThiagoBezzy** [comment]
Não estou conseguindo conectar também, o erro é o seguinte

Traceback (most recent call last):
  File "/usr/share/emesene/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'

**Comment by ghost** [comment]
Same problem here. Debian squeeze x86-64, emesene 1.6.3. Pidgin and aMSN work well.

**Comment by Sbte** [comment]
Should be fixed with current master.

**Comment by moneycat** [comment]
Confirm the solution with "local-bay" via modifying the two files. Thanks for developers.

**Comment by ghost** [comment]
Is there any solution for 1.6.3 ? I've changed all "contacts.msn.com" into "local-bay.contacts.msn.com" but no use.

<pre>
soap.manager.process(): &lt;SoapRequest FindMembership (400, 'Bad Request')&gt;
soap.manager.process(): Calling onGetMembershipList()
soap.manager.process(): &lt;SoapRequest ABFindAll (400, 'Bad Request')&gt;
soap.manager.process(): Calling onGetAddressBook()
</pre>

**Comment by Speppa** [comment]
Hello, I've run the portable version of this archive https://github.com/emesene/emesene/zipball/master , but it cannot connect to the msn account:

ubuntu@ubuntu:~/emesene-emesene-79c36ab/emesene$ ./emesene
No handlers could be found for logger "emesene"
Traceback (most recent call last):
  File "/home/ubuntu/emesene-emesene-79c36ab/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'

and here it's blocked, username e password are correct.
I've tried also the version 2.11.9 from the tarball and it has the same problem:

ubuntu@ubuntu:~/emesene-emesene-93a12ab/emesene$ ./emesene 
/home/ubuntu/emesene-emesene-93a12ab/emesene/e3/jabber/Worker.py:27: DeprecationWarning: the sha module is deprecated; use the hashlib module instead
  import sha
No handlers could be found for logger "emesene"
Traceback (most recent call last):
  File "/home/ubuntu/emesene-emesene-93a12ab/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'

I've run them on ubuntu 10.04 and opensuse 11.4.

**Comment by Sbte** [comment]
I think papyon needs to redirect through byrdr.omega.contacts.msn.com, like all clients that still work do.

**Comment by danielgomon** [comment]
The git version it's working ok right now!! Thanks!! :)

**Comment by hussani** [comment]
I'm using the portable version.
It's working, but the style of conversation window is broken.

**Comment by bardok** [comment]
Not work for me under xubuntu 11.10 (voyager) with emesene-emesene-79c36ab

**Comment by rflobao** [comment]
I have two msn accounts. The problem only exists in one of the accounts.
I'm using the latest update via git.

**Comment by luqmana** [comment]
Yep, I'm getting the same issue here on Ubuntu 11.10 64bit as well:
```-> % ./emesene -v  
No handlers could be found for logger "emesene"
[18:18:10 INFO papyon.msnp2p.transport] Registering transport <NotificationP2PTransport object at 0x2452640 (papyon+msnp2p+transport+base+BaseP2PTransport at 0x261acc0)>
[18:18:14 INFO papyon.service.address_book] [???] Received sync request:
...contacts: []
...groups: []
...memberships: []
Traceback (most recent call last):
  File "/home/luqman/Develop/emesene/emesene/e3/papylib/papyon/papyon/msnp/notification.py", line 796, in _address_book_state_changed_cb
    self._client.profile._cid = address_book.profile._cid
AttributeError: 'NoneType' object has no attribute '_cid'

```

I also can't connect with Empathy (from the empathy team ppa or from main ubuntu repos).
```

**Comment by slipky** [comment]
same here, ubuntu 10.04 64bit..

**Comment by rodrigozr** [comment]
To those still having problems... download the latest papyon and copy it to "emesene/e3/papylib/papyon".
http://www.freedesktop.org/wiki/Software/papyon
To download using git:
git clone git://anongit.freedesktop.org/papyon

Cheers!

**Comment by Sbte** [comment]
@rodrigozr, that version doesn't work properly either...

**Comment by rodrigozr** [comment]
@Sbte, weird... worked fine for me. I updated first to the lates emesene version, then copied the new papyon on top of it.

**Comment by Sbte** [comment]
@rodrigozr, the papyon code sometimes redirects to a server that can't be connected to, so my guess is that it works for the same amount of people the other fix that is currently in emesene works for. Both fixes might work for different people though.

**Comment by ghost** [comment]
Is there any solution about this issue
I tried the fix on emesene Blog but not work
I also tried the newest version of papyon but nothing happened

 Is in progress the issue ? will be an update soon on Ubuntu 11.04 ?

**Comment by Sbte** [comment]
The version you can download here should work: https://github.com/emesene/emesene/zipball/master

Just extract and run `./emesene/emesene`

If that doesn't work, please run `./emesene/emesene -vvv` and post the logs somewhere so we can read them.

**Comment by ghost** [comment]
@Sbte yess indeed it work :)
Thanks for the help-reply
Will be soon an update , or i have to run the command always in order to login ? :)

**Comment by Sbte** [comment]
@ALdaperan, we will release a new version as soon as possible, and as well update the ppa and release a fixed version into the official ubuntu repositories as SRU.

**Comment by ghost** [comment]
thank you Sbte :)

**Comment by mjtrangoni** [comment]
That with papyon from git and then copying works fine on Debian Wheezy! Thanks

**Comment by ghost** [comment]
Is there any fix for emesene 1.6.3 ?
I found this https://github.com/csuarez/emesene-1.6.3-fixed but i dont know how to make it work ?
Please help , i love emesene 1.6.3
